src/cmd/manage_nsfs.js

```diff
@@ -215,40 +215,29 @@ async function add_bucket(data) {
 }
 
 /** verify_bucket_owner will check if the bucket_owner has an account
+ * bucket_owner is the account name in the account schema
  * after it finds one, it returns the account id, otherwise it would throw an error
  * (in case the action is add bucket it also checks that the owner has allow_bucket_creation)
- * @param {string} bucket_owner account name
+ * @param {string} bucket_owner
  * @param {string} action
  */
 async function verify_bucket_owner(bucket_owner, action) {
-    let is_bucket_owner_exist = false;
-    let is_allow_bucket_creation = false;
-    let account_id;
-    const show_secrets = false;
-    const fs_context = native_fs_utils.get_process_fs_context();
-    const entries = await nb_native().fs.readdir(fs_context, accounts_dir_path);
-    // Gap - should replace this implementation
-    // it keeps iterating even if we find that the bucket owner exist
-    await P.map_with_concurrency(10, entries, async entry => {
-        if (entry.name.endsWith('.json') && !is_bucket_owner_exist) {
-            const full_path = path.join(accounts_dir_path, entry.name);
-            const data = await get_config_data(full_path, show_secrets);
-            if (data.name === bucket_owner) {
-                is_bucket_owner_exist = true;
-                is_allow_bucket_creation = data.allow_bucket_creation;
-                account_id = data._id;
-            }
-        }
-    });
-
+    // check if bucket owner exists
+    const fs_context = native_fs_utils.get_process_fs_context(config_root_backend);
+    const account_config_path = get_config_file_path(accounts_dir_path, bucket_owner);
+    const is_bucket_owner_exist = await native_fs_utils.is_path_exists(fs_context, account_config_path);
```

Comment:
please try to avoid using both is_path_exists() and get_config_data() to the same path,
we can reduce these 2 FS calls to 1 to increase performance

Reply:
Thank you